pkg/resources/grant_database_role.go

```diff
@@ -153,44 +153,10 @@ func ReadGrantDatabaseRole(d *schema.ResourceData, meta interface{}) error {
 
 	var found bool
 	for _, grant := range grants {
-		if grant.GrantedTo == sdk.ObjectType(objectType) {
-			if grant.GrantedTo == sdk.ObjectTypeRole || grant.GrantedTo == sdk.ObjectTypeShare {
-				if grant.GranteeName.FullyQualifiedName() == targetIdentifier {
-					found = true
-					break
-				}
-			} else {
-				/*
-					note that grantee_name is not saved as a valid identifier in the
-					SHOW GRANTS OF DATABASE ROLE <database_role_name> command
-					for example, "ABC"."test_parent_role" is saved as ABC."test_parent_role"
-					or "ABC"."test_parent_role" is saved as ABC.test_parent_role
-					and our internal mapper thereby fails to parse it correctly, returning "ABC."test_parent_role"
-					so this funny string replacement is needed to make it work
-				*/
-				s := grant.GranteeName.FullyQualifiedName()
-				if !strings.Contains(s, "\"") {
-					parts := strings.Split(s, ".")
-					s = sdk.NewDatabaseObjectIdentifier(parts[0], parts[1]).FullyQualifiedName()
-				} else {
-					parts := strings.Split(s, "\".\"")
-					if len(parts) < 2 {
-						parts = strings.Split(s, "\".")
-						if len(parts) < 2 {
-							parts = strings.Split(s, ".\"")
-							if len(parts) < 2 {
-								s = strings.Trim(s, "\"")
-								parts = strings.Split(s, ".")
-							}
-						}
-					}
-					s = sdk.NewDatabaseObjectIdentifier(parts[0], parts[1]).FullyQualifiedName()
-				}
-				if s == targetIdentifier {
-					found = true
-					break
-				}
-			}
+		if grant.GrantedTo == sdk.ObjectType(objectType) &&
```

Comment:
I'm not sure, but I believe this may not work with quoted database and unquoted database role, e.g. `"database".ROLE`.
Let's say we have database role id "database".ROLE.
NewAccountObjectIdentifier is trimming quotes so we end up with database".ROLE, then in the code above you're comparing some id with FQN which will print "database".ROLE" (I had this issue today 😠)

Kk I see it's handled below

Reply:
Actually, it was not handled properly :) I changed the logic a bit. I decided to parse it entirely in Show because of the quoting issues you mentioned.